x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -305,60 +330,75 @@ export const ComplianceDashboard = () => {
     enabled: isCloudSecurityPostureInstalled,
   });
 
-  useEffect(() => {
-    if (hasUserSelectedTab) {
-      return;
+  const location = useLocation();
+  const history = useHistory();
+  const { services } = useKibana();
+
+  const currentTabUrlState: PosturePolicyTemplate | undefined = useMemo(() => {
+    let tab: PosturePolicyTemplate | undefined;
+    if (location.pathname === cloudPosturePages.kspm_dashboard.path) {
+      tab = POSTURE_TYPE_KSPM;
     }
 
-    const preferredDashboard = getDefaultTab(
-      getSetupStatus,
-      getCspmDashboardData.data,
-      getKspmDashboardData.data
-    );
-    setSelectedTab(preferredDashboard);
-  }, [
-    getCspmDashboardData.data,
-    getCspmDashboardData.data?.stats.totalFindings,
-    getKspmDashboardData.data,
-    getKspmDashboardData.data?.stats.totalFindings,
-    getSetupStatus,
-    getSetupStatus?.cspm?.status,
-    getSetupStatus?.kspm?.status,
-    hasUserSelectedTab,
-  ]);
+    if (location.pathname === cloudPosturePages.cspm_dashboard.path) {
+      tab = POSTURE_TYPE_CSPM;
+    }
 
-  const tabs = useMemo(
-    () =>
-      isCloudSecurityPostureInstalled
-        ? [
-            {
-              label: i18n.translate('xpack.csp.dashboardTabs.cloudTab.tabTitle', {
-                defaultMessage: 'Cloud',
-              }),
-              'data-test-subj': CLOUD_DASHBOARD_TAB,
-              isSelected: selectedTab === POSTURE_TYPE_CSPM,
-              onClick: () => {
-                setSelectedTab(POSTURE_TYPE_CSPM);
-                setHasUserSelectedTab(true);
-              },
-              content: <TabContent posturetype={POSTURE_TYPE_CSPM} />,
+    // if the location is /dashboard or cloudPosturePages.dashboard.path, then return undefined
+    return tab;
+  }, [location.pathname]);
+
+  const preferredTabUrlState = useMemo(
+    () => getDefaultTab(getSetupStatus, getCspmDashboardData.data, getKspmDashboardData.data),
+    [getCspmDashboardData.data, getKspmDashboardData.data, getSetupStatus]
+  );
+
+  const tabs = useMemo(() => {
+    const navigateToPostureTypeDashboardTab = (pathname: string) => {
```

Comment:
this should probably be a hook in a separate, file next to `use_navigate_findings`
but also `navigateToPostureTypeDashboardTab` is just a nav to what ever path is being passed to it. which makes it a bit random. i would recommand to change the implementation for something like:

```
const navigateToPostureTypeDashboardTab = (postureType: 'kspm' | 'cspm') => {
   history.push(
      generatePath(benchmarksNavigation.rules.path, {
        postureType // this is assuming you are adding the postureType as optional param to the path, but if not you can also just do an "if" and navigate to the selected tab based on the value of postureType
      })
    );
}
```

Reply:
I tried `use_navigate_findings pattern` hooks. TabContent onClick Handler behaves differently than React.jsx element click handler. `onclick` does support  hook  being called